android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -145,6 +145,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
         searchView.setQuery(searchText, false);
         return true;
     }
+    @Override
+    public void onBackPressed(){
+        if((mSessions.isEmpty())&&(sessionRecyclerView.getVisibility()==View.GONE)){
```

Comment:
Yes, when on the map fragment, if we press backbitten and there are no sessions, the recycler view doesn't appear, but what the current code doesn't do is to make the no sessions text view visible. I have tried to address that here.